libraries/AP_HAL_ChibiOS/hwdef/PyroF4OSD/hwdef.dat

```diff
@@ -0,0 +1,153 @@
+# hw definition file for PyroDrone F4 OSD
+# WORK IN PROGRESS (NOT TESTED)
+
+# MCU class and specific type
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID 125
+
+# crystal frequency
+OSCILLATOR_HZ 8000000
+
+define STM32_ST_USE_TIMER 4
+define CH_CFG_ST_RESOLUTION 16
+
+FLASH_SIZE_KB 1024
+
+
+# only one I2C bus
+I2C_ORDER I2C1
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 UART2 USART3 UART4 UART5
+
+# LEDs
+PB9 LED_BLUE OUTPUT LOW GPIO(0)
+
+define HAL_GPIO_A_LED_PIN 0
+
+# buzzer
+PB5 BUZZER OUTPUT GPIO(80) LOW
+define HAL_BUZZER_PIN 80
+define HAL_BUZZER_ON 1
+define HAL_BUZZER_OFF 0
+
+# spi1 bus for IMU
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+
+# spi2 for OSD
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+PB12 MAX7456_CS CS
+PA4 MPU6000_CS CS
+
+# only one I2C bus in normal config
+PB6 I2C1_SCL I2C1
+PB7 I2C1_SDA I2C1
+
+# analog pins
+PC2 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+# define default battery setup
+# TODO: Figure this out
+# PC5 - ADC12_CH15
+define HAL_BATT_VOLT_PIN 15
+# PC4 - ADC12_CH14
+define HAL_BATT_CURR_PIN 14
+define HAL_BATT_VOLT_SCALE 10.1
+define HAL_BATT_CURR_SCALE 17.0
+
+# USART1
+PA9 USART1_TX USART1
+PA10 USART1_RX USART1
+
+# RC input using timer
+PA3 TIM9_CH2 TIM9 RCININT PULLDOWN
+
+# alternative RC input using UART
+# TODO: Compile error because of this?
+PA2 USART2_TX USART2 NODMA
```

Comment:
... but USART2 here (they are different)

Reply:
That fixed one of the compile issues, thanks for catching that. I'll see if I can find a definitive chart on which ones of these are actually USARTs, although most of the serial ports from the MatekF405 lined up with this pinout so I'm pretty sure they're correct. Is there a way to test which one is a USART?

Reply:
No idea :)